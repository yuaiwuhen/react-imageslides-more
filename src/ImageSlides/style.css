@keyframes line-scale-pulse-out {
  0% {
    transform: scaleY(1);
  }

  50% {
    transform: scaleY(0.4);
  }

  100% {
    transform: scaleY(1);
  }
}

.image-slides-loading {
  position: absolute;
  line-height: 500px;

  > div {
    display: inline-block;
    width: 4px;
    height: 35px;
    margin: 2px;
    border-radius: 2px;
    background-color: #fff;
    animation: line-scale-pulse-out 0.9s -0.6s infinite cubic-bezier(0.85, 0.25, 0.37, 0.85);
    animation-fill-mode: both;

    &:nth-child(2),
    &:nth-child(4) {
      animation-delay: -0.4s !important;
    }

    &:nth-child(1),
    &:nth-child(5) {
      animation-delay: -0.2s !important;
    }
  }
}

.image-slides-view-port {
  overflow: hidden;
  width: 100%;
  height: 100%;
  background-color: #000;
  touch-action: none;
}

.image-slides-container {
  display: inline-block;
  position: relative;
  overflow: visible;
  width: 100%;
  height: 100%;
  white-space: nowrap;
  will-change: transform;
  touch-action: none;
}

.image-slides-index {
  position: absolute;
  right: 10%;
  bottom: 10%;
  z-index: 100;
  border-radius: 2px;
  background-color: rgba(0, 0, 0, 0.15);
  color: #fff;
  font-size: 14px;
}

.image-slides-close {
  position: absolute;
  top: 10px;
  right: 10px;
  z-index: 100;
  padding-top: 6px;
  border: 0;
  border-radius: 2px;
  outline: none;
  background-color: rgba(0, 0, 0, 0.15);
  cursor: pointer;
}

.image-slides-blackboard {
  display: inline-flex;
  position: relative;
  overflow: hidden;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  margin-right: 10px;
  text-align: center;
  vertical-align: middle;
}

.image-slides-content {
  will-change: transform;
  user-select: none;
}

.leftcursor {
    cursor: url(https://img.t.sinajs.cn/t6/style/images/common/pic_prev.cur), auto;
}
.rightcursor {
    cursor: url(https://img.t.sinajs.cn/t6/style/images/common/pic_next.cur), auto;
}
