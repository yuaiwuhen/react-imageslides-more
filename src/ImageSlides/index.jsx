import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import transform from 'css3transform';
import AlloyFinger from '../AlloyFinger';
import ImageController from '../ImageController';
import touchEmulator from '../utils/touchEmulator';
import Overlay from '../Overlay';
import './style.css';

const GUTTER_WIDTH = 10;
const SWIPE_TRIGGER = 50;

function preload(url) {
    if (url) {
        const loader = new Image();
        return new Promise((resolve, reject) => {
            loader.onload = resolve;
            loader.onerror = reject;
            loader.src = url;
        });
    }
    return null;
}

export default class ImageSlides extends PureComponent {
    static propTypes = {
        images: PropTypes.arrayOf(PropTypes.string),
        index: PropTypes.number,
        isOpen: PropTypes.bool,
        isPageButton: PropTypes.bool,
        noTapClose: PropTypes.bool,
        useTouchEmulator: PropTypes.bool,
        addon: PropTypes.func,
        onClose: PropTypes.func,
        onChange: PropTypes.func,
        onNext: PropTypes.func,
        onPre: PropTypes.func,
    };
    
    static defaultProps = {
        images: [],
        index: 0,
        noTapClose: false,
        isOpen: false,
        useTouchEmulator: false,
    };
    state = {
        index: 0,
        haveControl: true,
        isOpen: false,
    };
    
    lastContainerOffsetX = 0;
    initialStyle = {};
    imageController = {};
    
    constructor(props) {
        super(props);
        const {index, isOpen} = props;
        this.state = {
            index: index || this.state.index,
            haveControl: false,
            isOpen,
        };
    }
    
    componentDidMount() {
        const {
            images,
        } = this.props;
        const {
            index,
        } = this.state;
        preload(images[index]);
        preload(images[index + 1]);
        preload(images[index - 1]);
    }
    
    componentWillReceiveProps(newProps) {
        const {isOpen} = this.props;
        const index = this.state.index;
        const {index: newIndex, isOpen: newIsOpen} = newProps;
        if (isOpen !== newIsOpen) {
            this.setState({
                isOpen: newIsOpen,
            });
        }
        if (newIndex !== undefined && index !== newIndex) {
            this.setState({
                index: newIndex,
            });
        }
    }
    
    getContainer = el => {
        if (el) {
            const {useTouchEmulator} = this.props;
            transform(el);
            this.containerEl = el;
            if (useTouchEmulator) {
                touchEmulator(el);
            }
        }
    };
    
    getControl = () => {
        if (!this.state.haveControl) {
            this.setState({
                haveControl: true,
            });
        }
    };
    
    handleContainerMove = e => {
        window.requestAnimationFrame(this.move(e));
        e.preventDefault();
        e.stopPropagation();
    };
    
    move = e => () => {
        if (this.state.haveControl) {
            this.containerEl.translateX = this.containerEl.translateX + parseInt(e.deltaX, 10);
        }
    };
    
    handleTouchEnd = e => {
        e.preventDefault();
        if (this.state.haveControl) {
            const {
                onChange,
                images,
            } = this.props;
            const boardWidth = (GUTTER_WIDTH + window.innerWidth);
            const baseline = boardWidth * this.getMedianIndex();
            if (-this.containerEl.translateX - baseline > SWIPE_TRIGGER) {
                const step = this.transition(160, 'next');
                if (onChange && this.state.index < images.length - 1) {
                    onChange(this.state.index + 1);
                }
                window.requestAnimationFrame(step);
            } else if (baseline + this.containerEl.translateX > SWIPE_TRIGGER) {
                const step = this.transition(160, 'last');
                if (onChange && this.state.index > 0) {
                    onChange(this.state.index - 1);
                }
                window.requestAnimationFrame(step);
            } else {
                const step = this.transition(160, 'noMove');
                window.requestAnimationFrame(step);
            }
            this.setState({
                haveControl: false,
            });
        }
    };
    
    handleNextPic = () => {
        const boardWidth = (GUTTER_WIDTH + window.innerWidth);
        const step = this.transition(160, 'next');
        window.requestAnimationFrame(step);
        this.setState({
            haveControl: false,
        });
    };
    
    handlePrevPic = () => {
        const boardWidth = (GUTTER_WIDTH + window.innerWidth);
        const step = this.transition(160, 'last');
        window.requestAnimationFrame(step);
        this.setState({
            haveControl: false,
        });
    };
    
    transition(time, direction) {
        const boardWidth = (GUTTER_WIDTH + window.innerWidth);
        let startTime;
        const startPos = this.containerEl.translateX;
        const size = (this.state.index === 0 && direction === 'last')
        || (this.state.index === this.props.images.length - 1 && direction === 'next')
        || direction === 'noMove' ? 0 : 1;
        const step = timestamp => {
            if (!startTime) startTime = timestamp;
            const progress = timestamp - startTime;
            this.containerEl.translateX = parseInt(startPos
                + progress / time * (-boardWidth * (this.getMedianIndex() + (direction === 'next' ? size : -size)) - startPos), 10);
            if (progress < time) {
                window.requestAnimationFrame(step);
            } else if (direction === 'next') {
                this.next();
            } else if (direction === 'last') {
                this.last();
            } else {
                this.updatePosition();
            }
        };
        return step;
    }
    
    
    getMedianIndex() {
        const {index} = this.state;
        const {images} = this.props;
        const displayMax = index + 2 > images.length ? images.length : index + 2;
        const displayMin = index - 1 < 0 ? 0 : index - 1;
        let center = parseInt((displayMax - displayMin) / 2, 10);
        if (index < 1) {
            center = index;
        } else if (index > images.length - 2) {
            center = images.length - index;
        }
        return center;
    }
    
    updatePosition = () => {
        this.containerEl.translateX = -(GUTTER_WIDTH + window.innerWidth) *
            this.getMedianIndex();
    };
    
    next() {
        const {index} = this.state;
        const {onNext, onPre, images} = this.props;
        if (index == images.length - 1) {
            onNext();
        }
        if (index < images.length - 1) {
            preload(images[index + 2]);
            this.setState(
                {
                    index: index + 1,
                },
                this.updatePosition,
            );
        } else {
            this.updatePosition();
        }
    }
    
    last() {
        const {index} = this.state;
        const {onNext, onPre, images} = this.props;
        if (index == 0) {
            onPre();
        }
        if (index > 0) {
            preload(images[index - 2]);
            this.setState(
                {
                    index: index - 1,
                },
                this.updatePosition,
            );
        } else {
            this.updatePosition();
        }
    }
    
    handleCloseViewer = e => {
        const {index} = this.state;
        const {onClose} = this.props;
        this.setState({
            isOpen: false,
        });
        if (onClose) onClose(e, index);
    };
    
    render() {
        const {index, isOpen} = this.state;
        const {images, addon, noTapClose, isPageButton} = this.props;
        const displayMax = index + 2 > images.length ? images.length : index + 2;
        const displayMin = index - 1 < 0 ? 0 : index - 1;
        return isOpen ? (
            <Overlay onClose={this.onCloseViewer}>
                <div className="image-slides-view-port">
                    {images.length > 0 && (
                        <div className="image-slides-index">
                            {`${index + 1} / ${images.length}`}
                        </div>
                    )}
                    {addon && addon(images[index], index, this.handleCloseViewer)}
                    <AlloyFinger
                        onSingleTap={noTapClose ? null : this.handleCloseViewer}
                        onTouchEnd={this.handleTouchEnd}
                        onPressMove={this.handleContainerMove}>
                        <div
                            className="image-slides-container"
                            ref={this.getContainer}>
                            {images.slice(displayMin, displayMax).map((url, ind) => (
                                /* eslint-disable */
                                <ImageController
                                    onGiveupControl={this.getControl}
                                    onNextPic={this.handleNextPic}
                                    onPrevPic={this.handlePrevPic}
                                    isPageButton={isPageButton}
                                    url={url}
                                    key={url + (ind + displayMin)}
                                    focused={ind === this.getMedianIndex()}/>
                            ))}
                        </div>
                    </AlloyFinger>
                </div>
            </Overlay>
        ) : null;
    }
}
