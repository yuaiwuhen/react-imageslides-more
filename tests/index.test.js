import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import ImageSlides from '../src/ImageSlides';

Enzyme.configure({ adapter: new Adapter() });
const images = [
  '',
  'http://img.zcool.cn/community/0101f856cfff206ac7252ce6214470.jpg',
  'http://a4.att.hudong.com/05/55/01200000033533115855502090905.jpg',
  'http://img.zcool.cn/community/0101f856cfff206ac7252ce6214470.jpg',
  'https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1503235534249&di=4c198d5a305627d12e5dae4c581c9e57&imgtype=0&src=http%3A%2F%2Fimg2.niutuku.com%2Fdesk%2Fanime%2F0529%2F0529-17277.jpg',
];

describe('<ImageSlides />', () => {
  it('should render a loading', () => {
    const wrapper = shallow(<ImageSlides images={images} isOpen />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders two imgs when then index is equal to 0', () => {
    const wrapper = shallow(<ImageSlides images={images} isOpen />);
    expect(wrapper.find('.image-slides-blackboard').length).toBe(2);
  });
  it('renders three imgs when then index is equal to 2', () => {
    const wrapper = shallow(<ImageSlides images={images} index={2} isOpen />);
    expect(wrapper.find('.image-slides-blackboard').length).toBe(3);
  });
});
