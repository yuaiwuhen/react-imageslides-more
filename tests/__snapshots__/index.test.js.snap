// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageSlides /> should render a loading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageSlides
    addon={null}
    images={
      Array [
        "",
        "http://img.zcool.cn/community/0101f856cfff206ac7252ce6214470.jpg",
        "http://a4.att.hudong.com/05/55/01200000033533115855502090905.jpg",
        "http://img.zcool.cn/community/0101f856cfff206ac7252ce6214470.jpg",
        "https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1503235534249&di=4c198d5a305627d12e5dae4c581c9e57&imgtype=0&src=http%3A%2F%2Fimg2.niutuku.com%2Fdesk%2Fanime%2F0529%2F0529-17277.jpg",
      ]
    }
    index={0}
    isOpen={true}
    onChange={null}
    onClose={null}
    useTouchEmulator={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="image-slides-view-port"
      >
        <div
          className="image-slides-index"
        >
          1 / 5
        </div>
        <div
          className="image-slides-container"
          style={
            Object {
              "transform": "translate3d(10px, 0, 0)",
            }
          }
        >
          <div
            className="image-slides-blackboard"
          >
            <div
              className="image-slides-loading"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </div>
          <div
            className="image-slides-blackboard"
          >
            <div
              className="image-slides-loading"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </div>
        </div>
      </div>,
      "onClose": [Function],
      "parentSelector": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image-slides-index"
          >
            1 / 5
          </div>,
          null,
          <div
            className="image-slides-container"
            style={
              Object {
                "transform": "translate3d(10px, 0, 0)",
              }
            }
          >
            <div
              className="image-slides-blackboard"
            >
              <div
                className="image-slides-loading"
              >
                <div />
                <div />
                <div />
                <div />
                <div />
              </div>
            </div>
            <div
              className="image-slides-blackboard"
            >
              <div
                className="image-slides-loading"
              >
                <div />
                <div />
                <div />
                <div />
                <div />
              </div>
            </div>
          </div>,
        ],
        "className": "image-slides-view-port",
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1 / 5",
            "className": "image-slides-index",
          },
          "ref": null,
          "rendered": "1 / 5",
          "type": "div",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="image-slides-blackboard"
              >
                <div
                  className="image-slides-loading"
                >
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                </div>
              </div>,
              <div
                className="image-slides-blackboard"
              >
                <div
                  className="image-slides-loading"
                >
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                </div>
              </div>,
            ],
            "className": "image-slides-container",
            "style": Object {
              "transform": "translate3d(10px, 0, 0)",
            },
          },
          "ref": [Function],
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="image-slides-loading"
                >
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                </div>,
                "className": "image-slides-blackboard",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "loading",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div />,
                    <div />,
                    <div />,
                    <div />,
                    <div />,
                  ],
                  "className": "image-slides-loading",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "http://img.zcool.cn/community/0101f856cfff206ac7252ce6214470.jpg1",
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="image-slides-loading"
                >
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                </div>,
                "className": "image-slides-blackboard",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "loading",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div />,
                    <div />,
                    <div />,
                    <div />,
                    <div />,
                  ],
                  "className": "image-slides-loading",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="image-slides-view-port"
        >
          <div
            className="image-slides-index"
          >
            1 / 5
          </div>
          <div
            className="image-slides-container"
            style={
              Object {
                "transform": "translate3d(10px, 0, 0)",
              }
            }
          >
            <div
              className="image-slides-blackboard"
            >
              <div
                className="image-slides-loading"
              >
                <div />
                <div />
                <div />
                <div />
                <div />
              </div>
            </div>
            <div
              className="image-slides-blackboard"
            >
              <div
                className="image-slides-loading"
              >
                <div />
                <div />
                <div />
                <div />
                <div />
              </div>
            </div>
          </div>
        </div>,
        "onClose": [Function],
        "parentSelector": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="image-slides-index"
            >
              1 / 5
            </div>,
            null,
            <div
              className="image-slides-container"
              style={
                Object {
                  "transform": "translate3d(10px, 0, 0)",
                }
              }
            >
              <div
                className="image-slides-blackboard"
              >
                <div
                  className="image-slides-loading"
                >
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                </div>
              </div>
              <div
                className="image-slides-blackboard"
              >
                <div
                  className="image-slides-loading"
                >
                  <div />
                  <div />
                  <div />
                  <div />
                  <div />
                </div>
              </div>
            </div>,
          ],
          "className": "image-slides-view-port",
        },
        "ref": [Function],
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1 / 5",
              "className": "image-slides-index",
            },
            "ref": null,
            "rendered": "1 / 5",
            "type": "div",
          },
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="image-slides-blackboard"
                >
                  <div
                    className="image-slides-loading"
                  >
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                </div>,
                <div
                  className="image-slides-blackboard"
                >
                  <div
                    className="image-slides-loading"
                  >
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                </div>,
              ],
              "className": "image-slides-container",
              "style": Object {
                "transform": "translate3d(10px, 0, 0)",
              },
            },
            "ref": [Function],
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="image-slides-loading"
                  >
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>,
                  "className": "image-slides-blackboard",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "loading",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div />,
                      <div />,
                      <div />,
                      <div />,
                      <div />,
                    ],
                    "className": "image-slides-loading",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "http://img.zcool.cn/community/0101f856cfff206ac7252ce6214470.jpg1",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="image-slides-loading"
                  >
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>,
                  "className": "image-slides-blackboard",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "loading",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div />,
                      <div />,
                      <div />,
                      <div />,
                      <div />,
                    ],
                    "className": "image-slides-loading",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
